tree/ntuple/v7/test/CustomStruct.hxx

```diff
@@ -90,4 +90,11 @@ struct StructWithEnums : BaseOfStructWithEnums {
    int b = static_cast<int>(DeclEC::E42);
 };
 
+/// A class that behaves as a collection accessed through the `TVirtualCollectionProxy` interface
+template <typename T>
+struct StructUsingCollectionProxy {
+   using ValueType = T;
+   std::vector<T> v; //! do not accidentally store via RClassField
```

Comment:
This `std::vector<T>` _can_ be also directly stored/retrieved via `RClassField`; the member is tagged as transient to avoid the unit test from passing if, for some reason, the wrong `RField` specialization is instantiated.

Now that we are also throwing if `RClassField` is used for classes with an associated collection proxy, this may (or not) be deleted.

Reply:
In principle, I will leave the `//!` marker there, as it should make the test fail _even if the logic somewhere else is severely broken_, i.e. either storage is handled via the collection proxy or no handled at all.